src/plugins/dashboard/public/application/listing/dashboard_unsaved_listing.tsx

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import React, { useCallback, useEffect, useState } from 'react';
+import useMount from 'react-use/lib/useMount';
+import { DashboardSavedObject } from '../..';
+import { dashboardUnsavedListingStrings, getNewDashboardTitle } from '../../dashboard_strings';
+import { useKibana } from '../../services/kibana_react';
+import { DASHBOARD_PANELS_UNSAVED_ID } from '../lib/dashboard_panel_storage';
+import { DashboardAppServices, DashboardRedirect } from '../types';
+import { confirmDiscardUnsavedChanges } from './confirm_overlays';
+
+const DashboardUnsavedItem = ({
+  dashboard,
+  onOpenClick,
+  onDiscardClick,
+}: {
+  dashboard?: DashboardSavedObject;
+  onOpenClick: () => void;
+  onDiscardClick: () => void;
+}) => {
+  return (
+    <>
+      <EuiTitle size="xxs" className="dshUnsavedListingItemTitle">
+        <h3 className="dshUnsavedListingItemTitle">
+          <EuiIcon type="pencil" /> {dashboard?.title ?? getNewDashboardTitle()}
+        </h3>
+      </EuiTitle>
+      <EuiFlexGroup gutterSize="s" alignItems="flexStart" className="dshUnsavedListingButtons">
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            flush="left"
+            size="xs"
+            color="primary"
+            onClick={onOpenClick}
+            data-test-subj={`edit-unsaved-${dashboard?.id ?? DASHBOARD_PANELS_UNSAVED_ID}`}
+          >
```

Comment:
If possible, it'd be nice to add an `aria-label` on this and the discard button that includes the name of the dashboard in the button. For example: `aria-label="Continue Editing My Sweet Dashboard`

(This way, instead of _n_ possible buttons that all say "Continue editing" we get a unique button for each unique action.)

Reply:
good call thanks, implemented this. Let me know if you think of any other a11y things to add!